src/pretix/base/models/orders.py

```diff
@@ -229,7 +229,14 @@ def can_modify_answers(self) -> bool:
 
     @property
     def can_user_cancel(self) -> bool:
-        return self.event.settings.cancel_allow_user
+        positions = self.positions.all().prefetch_related('item')
+        cancelable = True
+        cancelable_products = []
+        for i, op in enumerate(positions):
+            cancelable_products.append(op.item.allow_cancel)
+        if False in cancelable_products:
```

Comment:
I believe this could be written in a more concise way that is still very readable, for example:

```
cancelable = not all([op.item.allow_cancel for op in positions])
```